das/daser.go

```diff
@@ -42,28 +61,49 @@ func (d *DASer) Start(context.Context) error {
 		return err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	go d.sampling(ctx, sub)
+	// load latest DASed checkpoint
+	checkpoint, err := loadCheckpoint(d.cstore)
+	if err != nil {
+		return err
+	}
+	log.Infow("loaded checkpoint", "height", checkpoint)
+
+	dasCtx, cancel := context.WithCancel(context.Background())
 	d.cancel = cancel
+	d.sampleDn, d.catchUpDn = make(chan struct{}), make(chan struct{})
```

Comment:
nit: you can set those in constructor and there is also no need to set those to nil in Stop. Setting cancel to nil to check if DASer was already started or not is enough. Also, we need to check if DASer was started or not actually, otherwise there are possible issues with those channels and cancel being set to nil.

Reply:
We do check if DASer was started though by checking to see if `d.cancel != nil` in `Start`

Reply:
Ok, true, not in this PR.

Reply:
Resolved https://github.com/celestiaorg/celestia-node/pull/473/commits/ad2975d10455ed92488bdbd81d5c30cf59e3a0bf ?